development-docs/TESTING.md

```diff
@@ -0,0 +1,186 @@
+# Testing Kafka Bridge
+
+This document gives a detailed breakdown of the testing processes and testing options for `Kafka Bridge` within tests.
+
+<!-- TOC depthFrom:2 -->
+
+- [Pre-requisites](#pre-requisites)
+- [Package Structure](#package-structure)
+- [Approaches](#approaches)
+- [Test Phases](#test-phases)
+- [Available Test Groups](#available-test-groups)
+- [Environment Variables](#environment-variables)
+- [Running single test class](#running-single-test-class)
+
+<!-- /TOC -->
+
+## Pre-requisites
+
+To run any tests you need build whole project. You can achieve that with following command  `mvn clean install -DskipTests` 
+in root directory of the project.
+
+The next requirement is to have [docker](https://docs.docker.com/get-docker/) installed, because we use test-containers. 
+In case of interest you can check our implementation of Strimzi `Kafka` container [here](https://github.com/strimzi/strimzi-kafka-operator/tree/master/test-container).
+
+## Package Structure
+
+You can find tests inside `src.test.java` package. Moreover we have the auxiliary classes and the most notable one are:
+ 
+- `src.test.java.io.strimzi.kafka.bridge.utils` = static methods
+- `src.test.java.io.strimzi.kafka.bridge.clients` = clients for testing overall communication of the `Kafka Bridge`
+- `src.test.java.io.strimzi.kafka.bridge.facades` = encapsulation of the standalone `Kafka` and `AdminClient` instance
```

Comment:
it's strange mixing folder path with package, I think we can avoid this and highlights only classes names `Utils`, `Clients` and `Facades` then using a list and not "=" :-)